servicetalk-opentelemetry-http/src/main/java/io/servicetalk/opentelemetry/http/AbstractOpenTelemetryHttpRequesterFilter.java

```diff
@@ -28,12 +28,17 @@
 import io.servicetalk.transport.api.ConnectionInfo;
 
 import io.opentelemetry.context.Context;
+import io.opentelemetry.context.ContextKey;
+import io.opentelemetry.context.Scope;
 
 import javax.annotation.Nullable;
 
 abstract class AbstractOpenTelemetryHttpRequesterFilter extends AbstractOpenTelemetryFilter
         implements StreamingHttpClientFilterFactory, StreamingHttpConnectionFilterFactory {
 
+    // package private for testing
+    static final ContextKey<Boolean> SHOULD_INSTRUMENT_KEY = ContextKey.named("should-instrument-carrier");
```

Comment:
heh, looks like OTEL made the same decision as our keys that only an instance matters, not the name :)